src/libasr/codegen/llvm_utils.h

```diff
@@ -96,6 +96,11 @@ namespace LFortran {
 
             void start_new_block(llvm::BasicBlock *bb);
 
+            template <typename IF, typename ELSE>
+            void create_if_else(llvm::Value * cond, IF if_block, ELSE else_block);
+            template <typename Cond, typename Body>
+            void create_loop(Cond condition, Body loop_body);
```

Comment:
Let's move all the implementation here.

And add documentation that this is an optional API, that does not have to be used, that many times for more complicated things it might be more readable to just use the LLVM API without any extra layer on top. In some other cases it might be more readable to use this abstraction. So this API is provided and can be optionally used, but it is not required.